### PR TITLE
[crypto] poseidon.js

### PR DESCRIPTION
**Pull request by mimoo**
This commit exports an OCaml poseidon hash implemention to javascript.
It uses the fp_3 parameters (see https://github.com/o1-labs/marlin/blob/master/oracle/src/poseidon.rs#L73).
In particular, the javascript implementation has two methods:

- `hashBytearray` to hash an opaque array of bytes. The type expected is a Uint8Array.
- `hashFieldElems` to hash an array of field elements. The field elements must be represented in little-endian hex encoded form.

Both methods return a digest in hexadecimal form of a little-endian encoded field element.

For more details see https://hackmd.io/ecBTibDpTzGdH9WM30gwQw

### CODE REVIEW

File: src/app/client_sdk/poseidon_hash.ml
Comment (mrmr1993) [inline src/app/client_sdk/poseidon_hash.ml:50]: Should this PR be against `develop`? I think we want to avoid including this in the client SDK until we've transitioned to this configuration for the main hash function too, which will be a breaking change that we'll want to land there.

**Review by mrmr1993 — COMMENTED** [review]
Code looks good, but left a comment about where we should land this PR.

File: src/app/client_sdk/poseidon_hash.ml
Comment (mimoo) [reply src/app/client_sdk/poseidon_hash.ml:50]: maybe a good time to ask: is there some documentation on how we manage different branches and create releases?

Comment (mrmr1993) [reply src/app/client_sdk/poseidon_hash.ml:50]: For branches, the current model is:
* anything that is protocol-compatible with the current mainnet goes in `compatible`
* anything that needs an associated hard-fork goes in `develop`

There's some more nuance about how and when to land things, and how the released work; I think there's a notion document somewhere with the details.

Comment (mimoo) [reply src/app/client_sdk/poseidon_hash.ml:50]: and is develop rebased on compatible periodically? Also what about master since it is the default branch?

Comment (mrmr1993) [reply src/app/client_sdk/poseidon_hash.ml:50]: Compatible should be merged into develop, ideally straight after merging each PR into compatible.

Comment (mimoo) [reply src/app/client_sdk/poseidon_hash.ml:50]: What's the conclusion on this :D ? I'm fine with either way

**Comment by mimoo** [comment]
rebased

**Review by mrmr1993 — APPROVED** [review]
Looks like this needs a reformat, but code LGTM

**Comment by mimoo** [comment]
!ci-build-me

**Comment by mimoo** [comment]
looks like we have to wait for https://github.com/MinaProtocol/mina/pull/9124 for this to build on develop

**Pull request merged**
